testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/asserts/AttributesAssert.groovy

```diff
@@ -58,12 +58,6 @@ class AttributesAssert {
     Set<String> unverifiedAttributes = new TreeSet(allAttributes)
     unverifiedAttributes.removeAll(assertedAttributes)
 
-    // Don't need to verify thread details.
-    assertedAttributes.add("thread.id")
```

Comment:
These were added by me when migrating the tests to run with the java agent, before we somehow skipped the thread processor. It's a pretty mysterious hack so glad they're gone.